scrnaseq.Rmd

```diff
@@ -294,30 +312,93 @@ sc = subset(sc, features=features_keep)
 sc
 ```
 
-## Normalisation, feature selection and scaling {.tabset}
-Feature selection: For downstream analysis it is beneficial to focus on genes that exhibit high cell-to-cell variation, that is they are highly expressed in some cells and lowly expressed in others. 
+## Normalisation, cell cycle scoring, scaling and variable features {.tabset}
+In this section, we subsequently run a series of Seurat functions:  
+1. We start by running a __standard log normalisation__, where counts for each cell are divided by the total counts for that cell and multiplied by 10,000. This is then natural-log transformed.   
+2. To be able to compare normalised gene counts between genes, gene counts are further __scaled__ to have zero mean and unit variance (z-score). This way, genes are equally weighted for downstream analysis.  
+3. __Variable features__ are selected. For downstream analysis it is beneficial to focus on genes that exhibit high cell-to-cell variation, that is they are highly expressed in some cells and lowly expressed in others. 
+
+In addition, we run:  
+4. __SCTransform__, a new and more sophisticated normalisation method that replaces the previous functions (__normalisation, scaling and variable features__). Note that although results are kept for steps 1-3 ("RNA" assay), downstream analyses will default to resulting data of step 4 ("SCT" assay). 
+
+If specified in the above parameter section, __cell cycle scores__ are assigned to each cell based on its normalised expression of G2/M and S phase markers, after basic normalisation (step 1). Cell cycle effects are then removed during scaling (step 4). Done for this report: `r param$cc_remove`. 
 
-Scaling: To be able to compare normalised gene counts between genes, gene counts are further scaled to have zero mean and unit variance (z-score). This way, genes are equally weighted for downstream analysis. 
 ```{r norm, results="hide"}
+# This function often crashes with multicore support, so stop for a moment
+future::plan("sequential")
+
 # Normalise data the original way
 sc = Seurat::NormalizeData(sc, normalization.method = "LogNormalize", scale.factor = 10000)
 
-# Select features from normalised data
-sc = Seurat::FindVariableFeatures(sc, selection.method = "vst", nfeatures = 2000)
+# Back to multicore support
+future::plan("multicore")
+```
+
+```{r cc_scores, eval=param$cc_remove}
+if(param$cc_remove) {
+
+# Use biomart to translate human cell cycle genes to the species of interest
+mart_human = biomaRt::useEnsembl("ensembl", dataset="hsapiens_gene_ensembl", mirror=mirror) 
```

Comment:
The translation for the cell cycle genes could be done in a previous chunk irrespective of whether we remove or not? It would be much clearer. Also, we may have other lists that we also need to translate to human

Reply:
I switched chunks, see next commit.